core/src/apps/cardano/sign_tx.py

```diff
@@ -490,27 +543,105 @@ async def _process_auxiliary_data(
     await ctx.call(auxiliary_data_supplement, CardanoTxHostAck)
 
 
-async def _process_witnesses(
+async def _process_minting(
+    ctx: wire.Context, builder: TxBuilder, has_token_minting: bool
+) -> None:
+    """Read, validate and serialize the asset groups of token minting."""
+    if not has_token_minting:
+        return
+
+    token_minting: CardanoTxMint = await ctx.call(CardanoTxItemAck(), CardanoTxMint)
+
+    await show_warning_tx_contains_mint(ctx)
+
+    # until the CIP with canonical CBOR is finalized storing the seen_policy_ids is the only way we can check for
+    # duplicate policy_ids
+    seen_policy_ids: set[bytes] = set()
+    builder.start_minting(token_minting.asset_groups_count)
+    for _ in range(token_minting.asset_groups_count):
+        asset_group: CardanoAssetGroup = await ctx.call(
+            CardanoTxItemAck(), CardanoAssetGroup
+        )
+        asset_group.policy_id = bytes(asset_group.policy_id)
```

Comment:
Weirdly it seems it is redundant :+1: I guess this changed either with tx streaming or Trezor changed something because protobuf `bytes` were previously received as `bytearray` I think - that's why the conversion was needed.

Reply:
https://github.com/vacuumlabs/trezor-firmware/pull/66/commits/ca9a89b2b753fbf0d744fb3bbec1335081ed6089#diff-eaac9c946526e07b3ac5c317971467dc330f41ee2cd7da91e701aa3098365bdbR560